### PR TITLE
cln-rpc: rewrite is_none_or_empty to not use clone()

### PR DESCRIPTION
**Pull request by daywalker90**
This should be the same, but without cloning.

### CODE REVIEW

File: cln-rpc/src/lib.rs [hunk 1]
```diff
@@ -130,9 +130,11 @@ fn is_none_or_empty<T>(f: &Option<Vec<T>>) -> bool
...
     T: Clone,
 {
-    // TODO Find a better way to check, possibly without cloning
-    let f = f.clone();
-    f.is_none() || f.unwrap().is_empty()
+    if let Some(inner) = f {
+        inner.is_empty()
+    } else {
+        true
+    }
```
Comment (vincenzopalazzo) [inline cln-rpc/src/lib.rs:137]:
```suggestion
    f.map_or(true, |value| value.is_empty())
```

I think this is a better way of writing it?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

**Comment by daywalker90** [comment]
even better, yes!

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/6868/commits/413945a47562794edafc1a0fa132fe05e7b467e6

**Review by vincenzopalazzo — APPROVED** [review]
Ok I took the freedom to fix the git history and rebased it.

This should not create CI failure, but lets wait

ACK https://github.com/ElementsProject/lightning/pull/6868/commits/5f19a889235793b685c5d61d5f2fb9d113ac31d9

**Review by vincenzopalazzo — APPROVED** [review]
Ok I was wrong because with the `f.map` take the ownership.

So better use `as_ref()` to fix the following error

```
➜  lightning git:(pr6868) ✗ cargo build
   Compiling cln-rpc v0.1.7 (/home/vincent/github/work/lightning/cln-rpc)
error[E0507]: cannot move out of `*f` which is behind a shared reference
    --> cln-rpc/src/lib.rs:133:5
     |
133  |     f.map_or(true, |value| value.is_empty())
     |     ^ -------------------------------------- `*f` moved due to this method call
     |     |
     |     help: consider calling `.as_ref()` or `.as_mut()` to borrow the type's contents
     |     move occurs because `*f` has type `std::option::Option<Vec<T>>`, which does not implement the `Copy` trait
     |
note: `std::option::Option::<T>::map_or` takes ownership of the receiver `self`, which moves `*f`
    --> /home/vincent/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:1129:25
     |
1129 |     pub fn map_or<U, F>(self, default: U, f: F) -> U
     |                         ^^^^
help: you can `clone` the value and consume it, but this might not be your desired behavior
     |
133  |     f.clone().map_or(true, |value| value.is_empty())
     |      ++++++++

For more information about this error, try `rustc --explain E0507`.
```

Now we are good 

ACK https://github.com/ElementsProject/lightning/pull/6868/commits/f6f296b940afc92e09c7226509b0851d0f82acc4

**Pull request merged**